src/Components/Web.JS/src/Boot.WebAssembly.ts

```diff
@@ -61,3 +63,19 @@ if (shouldAutoStart()) {
     }
   });
 }
+
+async function intiializeConfigAsync(platform: Platform, resourceLoader: WebAssemblyResourceLoader) : Promise<void> {
+  const configFiles = resourceLoader.readConfigFilesAsync();
+  const resolvedFiles = await Promise.all(configFiles.map(async c => {
+    const content = new Uint8Array(await c.contentPromise);
+    return { name: c.name, content: content};
+  }));
+
+
+  window['Blazor']._internal.getApplicationEnvironment = () => platform.toDotNetString(resourceLoader.applicationEnvironment);
+  window['Blazor']._internal.getConfig = (string: System_String) : Pointer | undefined => {
+    const fileName = platform.toJavaScriptString(string);
+    const resolvedFile = resolvedFiles.find(f => f.name == fileName);
+    return resolvedFile ? platform.toDotNetArray(resolvedFile.content) : undefined;
```

Comment:
Why do we want to allow the non-match condition? If it would be OK to let this be unsupported (which I suspect it should be), perhaps we could remove the logic for dealing with non-matches and let the function return type simply be `Pointer` (or better, `System_Array` if we still have that).

Reply:
Wait, I see. The .NET code explicitly asks for `appsettings.json` and `appsettings.envname.json` without knowing if they exist. So this makes sense.